src/core/AnimatedValue.js

```diff
@@ -14,21 +16,44 @@ export default class AnimatedValue extends AnimatedNode {
   }
 
   __detach() {
+    this.__detachAnimation();
     super.__detach();
   }
 
+  __detachAnimation(result) {
+    if (this.animation) {
+      this.animation.returnMethod &&
```

Comment:
let's call this `callback` or `animationCallback` rather than `returnMethod` (it does not return anything)